100.md

```diff
@@ -0,0 +1,552 @@
+
+## Methods
+
+- **get_public_key**
+  - params:
+
+    ```js
+    const intent = `intent:#Intent;scheme=nostrsigner;S.compressionType=none;S.returnType=signature;S.type=get_public_key;S.callbackUrl=https://example.com/?event=;end`;
+
+    window.href = intent;
+    ```
+
+- **sign_event**
+  - params:
+
+    ```js
+    const intent = `intent:${eventJson}#Intent;scheme=nostrsigner;S.compressionType=none;S.returnType=signature;S.type=sign_event;S.callbackUrl=https://example.com/?event=;end`;
+
+    window.href = intent;
+    ``` 
+
+- **nip04_encrypt**
+  - params:
+
+    ```js
+    const intent = `intent:${plainText}#Intent;scheme=nostrsigner;S.pubKey=${hex_pub_key};S.compressionType=none;S.returnType=signature;S.type=nip04_encrypt;S.callbackUrl=https://example.com/?event=;end`;
+
+    window.href = intent;
+    ``` 
+
+- **nip44_encrypt**
+  - params:
+
+    ```js
+    const intent = `intent:${plainText}#Intent;scheme=nostrsigner;S.pubKey=${hex_pub_key};S.compressionType=none;S.returnType=signature;S.type=nip44_encrypt;S.callbackUrl=https://example.com/?event=;end`;
+
+    window.href = intent;
+    ```   
+
+- **nip04_decrypt**
+  - params:
+
+    ```js
+    const intent = `intent:${encryptedText}#Intent;scheme=nostrsigner;S.pubKey=${hex_pub_key};S.compressionType=none;S.returnType=signature;S.type=nip04_decrypt;S.callbackUrl=https://example.com/?event=;end`;
+
+    window.href = intent;
+    ```     
+
+- **nip44_decrypt**
+  - params:
+
+    ```js
+    const intent = `intent:${encryptedText}#Intent;scheme=nostrsigner;S.pubKey=${hex_pub_key};S.compressionType=none;S.returnType=signature;S.type=nip44_decrypt;S.callbackUrl=https://example.com/?event=;end`;
+
+    window.href = intent;
+    ```     
+
+- **decrypt_zap_event**
+  - params:
+
+    ```js
+    const intent = `intent:${eventJson}#Intent;scheme=nostrsigner;S.compressionType=none;S.returnType=signature;S.type=decrypt_zap_event;S.callbackUrl=https://example.com/?event=;end`;
+
+    window.href = intent;
+    ```          
+
+## Example
+
+```js
+<!DOCTYPE html>
+<html lang="en">
+<head>
+    <meta charset="UTF-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <title>Document</title>
+</head>
+<body>
+    <h1>Test</h1>
+       
+    <script>
+        window.onload = function() {
+            var url = new URL(window.location.href);
+            var params = url.searchParams;
+            if (params) {
+                var param1 = params.get("event");
+                if (param1) alert(param1)
+            }
+            let json = {
+                kind: 1,
+                content: "test"
+            }
+            let encodedJson = encodeURIComponent(JSON.stringify(json))
+            var newAnchor = document.createElement("a");
+            newAnchor.href = `intent:${encodedJson}#Intent;scheme=nostrsigner;S.compressionType=none;S.returnType=signature;S.type=sign_event;S.callbackUrl=https://example.com/?event=;end`;
```

Comment:
This is notably an Android-centric URI. For example, iOS doesn't have intents. Does this mean that web applications need to start being platform aware? If so, I'm not sure if this is a good choice. Can we achieve the same result in a platform-agnostic way?

I've been thinking through how to build an iOS signer application, and I feel like we can shape this spec to be platform agnostic (while still allowing flexibility for platform-specific first-class features be used to make the UX better).

Something like
```js
newAnchor.href = `nostrsigner:${encodedJson}?compressionType=none&S.returnType=signature;S.type=sign_event&callbackUrl=https://example.com/?event=;end
```

Let me know if I've misunderstood anything.

Reply:
We can change to do the way you did so it is platform-agnostic.